include/RAJA/util/Allocator.hpp

```diff
@@ -0,0 +1,89 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   RAJA header file containing an implementation of a memory pool.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-21, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_UTIL_ALLOCATOR_HPP
+#define RAJA_UTIL_ALLOCATOR_HPP
+
+#include <cstddef>
+#include <string>
+#include <vector>
+
+#include "RAJA/util/camp_aliases.hpp"
+
+namespace RAJA
+{
+
+/*! \class Allocator
+ ******************************************************************************
+ *
+ * \brief  Allocator Provides a generic interface for allocation in RAJA
+ *
+ * Allocator& device_pool = RAJA::cuda::get_device_allocator();
+ *
+ * RAJA::cuda::set_device_allocator<SomeOtherDevicePoolAllocator>(args...);
+ *
+ ******************************************************************************
+ */
+struct Allocator
```

Comment:
Here is the Allocator class that is used to make memory allocation more flexible.